src/utils/promisifiedJWTVerify.ts

```diff
@@ -0,0 +1,22 @@
+import jwt from "jsonwebtoken";
+
+export function promisifiedJWTVerify<T>(oauthData: {
+  clientSecret?: string;
+  accessToken: string;
+}): Promise<T> {
+  const { clientSecret, accessToken } = oauthData;
+
+  return new Promise((resolve, reject) => {
+    if (clientSecret) {
+      jwt.verify(
+        accessToken,
+        clientSecret,
+        (error: jwt.VerifyErrors | null, decoded: any) => {
+          return error ? reject(error) : resolve(decoded);
+        },
+      );
+    } else {
+      reject("Missing client_secret");
```

Comment:
I would suggest doing something like

```ts
export class MissingClientSecretError extends Error{}

reject(new MissingClientSecretError("Missing client secret")
```

Reply:
Wait, in the end I don't need this if I implement the `Config` object, right?

Missing the `client_secret` will break the app.